### PR TITLE
Travis CI: Run doctests only on legacy Python

### LINKED ISSUE #525: Help enable doctest in .travis.yml
hi @cclauss 

Could you help enable doctest in `.travis.yml`?

### PR DESCRIPTION
**Pull request by cclauss**
Fixes #525 Once all tests pass, the __|| true__ at the end of __.travis.yml__ line 30 should be removed.  Also note that __experimental/pwt.py__ and __web/webopenid.py__ (#526) are explicitly ignored.

Comments of the form __# PY3DOCTEST: b'hello'__ have been inserted where we know what the Python 3 test should return.

### CODE REVIEW

**Comment by cclauss** [comment]
In the coverage report above:
1. Python 2 coverage is not much higher than Python 3 -- is it not counting the doctests?
2. Python 3.7 has no coverage report.

File: .travis.yml [hunk 1]
```diff
@@ -14,16 +14,21 @@ install:
...
   # exit-zero treats all errors as warnings.  The GitHub editor is 127 chars wide
   - flake8 . --count --exit-zero --max-complexity=10 --max-line-length=127 --statistics
   - psql -c 'create database webpy;' -U postgres
   - mysql --user=root -e "create user 'scott'@'localhost' identified by 'tiger'; create database webpy; grant all privileges on webpy.* to 'scott'@'localhost' with grant option;"
-script: pytest --cov --cov-report=term-missing --cov-report=xml
+script:
+  - pytest --cov --cov-report=term-missing --cov-report=xml
+  - if [ "$TRAVIS_PYTHON_VERSION" = "2.7" ]; then
+      pip install twisted ;
+      pytest --doctest-modules --ignore=experimental/pwt.py --ignore=web/webopenid.py . || true ;
```
Comment (iredmail) [inline .travis.yml]: We should run doctests for `web/*.py`, not `tests/*` and `experimental/*`.
Could you help fix this? Sorry i didn't make it clear in original issue.

Comment (cclauss) [reply .travis.yml]: Done.

**Comment by iredmail** [comment]
Some doctests are written for py3, but some for py2. Which one should we use for `0.40` stable release?

**Comment by cclauss** [comment]
Having all doctests working on Python 2 sounds like a good objective for 0.40.

Let's land this PR and then I will fix up all doctests to work on Python 2 (leaving comments about how to they should be modified for Python 3).  We should not wait long because we there are now < 100 days until Py2EoL

**Comment by iredmail** [comment]
Agree, let's get py2 working for 0.40.

Most failed doctests are just string types, for example (py2 + pytest):

```
_________________________________________________ [doctest] web.application.auto_application __________________________________________________
600     automatically using metaclass.
601 
602         >>> app = auto_application()
603         >>> class hello(app.page):
604         ...     def GET(self): return "hello, world"
605         ...
606         >>> class foo(app.page):
607         ...     path = '/foo/.*'
608         ...     def GET(self): return "foo"
609         >>> app.request("/hello").data
Expected:
    b'hello, world'
Got:
    'hello, world'
```

Replacing `b'hello, world'` by `'hello, world'` fixes the error.

**Comment by cclauss** [comment]
See the last commit

**Comment by cclauss** [comment]
OK...  I have doe what I could.

I commented out one doctest in __web/webapi.py__ and we __--ignore=web/template.py__  If @anandology has the secret sauce to fix those error then great.

**Pull request merged**